src/tasks/AndroidAppBuilder/Templates/MonoRunner.java

```diff
@@ -88,7 +90,8 @@ public static int initialize(String entryPointLibName, String[] args, Context co
         unzipAssets(context, filesDir, "assets.zip");
 
         Log.i("DOTNET", "MonoRunner initialize,, entryPointLibName=" + entryPointLibName);
-        return initRuntime(filesDir, cacheDir, testResultsDir, entryPointLibName, args);
+        int localDateTimeOffset = OffsetDateTime.now().getOffset().getTotalSeconds();
```

Comment:
Is there no way in Java to get the current TZ offset without calculating the current time?

Reply:
One issue is that all Android apps will be paying this cost of `OffsetDateTime.now()` here, even if the app doesn't use `DateTime.Now` in C# - especially in startup code.

Is there any "fast path" Java/Android API we could P/Invoke into in `GetLocalUtcOffset` instead?

Reply:
Actually, looking into this more, I'm not sure how this is correct to use `.now()` in the `TimeZoneInfo.GetLocalUtcOffset` method at all.

`TimeZoneInfo.GetLocalUtcOffset` takes the `DateTime` that we are interested in getting the offset for. Imagine if the first call passed in a `DateTime` from the winter and the app is currently running in the summer. That first call (for the first 1 seconds) is going to return the incorrect value. It is going to return an offset with daylight savings time, but the `DateTime` being requested isn't under daylight savings time.

Reply:
I think you're right. Since this performance issue was a problem with AndroidTZData taking a while to load, I tried to bump the implementation difference further down trying to account for all APIs that invoke AndroidTZData. However, it was decided offline that we can focus on `DateTimeOffset.Now` as it is more likely to be called at startup (compared to others), and other APIs would require a different information passed from Java to the mono runtime. Along the way I didn't pay enough attention to that `DateTimeOffset.Now` only calls one specific case of `GetLocalUtcOffset` when trying to get the background thread working. 

Since this PR is aimed at resolving the performance for `DateTimeOffset.Now`, would it be better to migrate `Now` into new `DateTimeOffset.cs` `DateTimeOffset.NonAndroid.cs` files and introduce the logic of fast path + background thread for the offset in there?

Reply:
Yeah, I'm realizing now that I conflated two different ideas from the original issue. I think @grendello mentioned that a P/Invoke + JNI would work, but might be slightly slower than the original ideas in https://github.com/dotnet/runtime/issues/71004#issuecomment-1172131137

Reply:
It would be enough if dotnet runtime handled the timezone change broadcast - or used one that [Xamarin.Android installs](https://github.com/xamarin/xamarin-android/blob/91f330f4a57af90c5f7bde223cf204b6afb5324f/src/java-runtime/java/mono/android/MonoPackageManager.java#L33-L36). My original suggestion was that Xamarin.Android would get the local offset somewhere [around here](https://github.com/xamarin/xamarin-android/blob/91f330f4a57af90c5f7bde223cf204b6afb5324f/src/java-runtime/java/mono/android/MonoPackageManager.java#L38-L44) and pass it to the Mono runtime when we call `monovm_inititialize_preparsed`. **If** within that time the tz would have changed, we would be able to capture that and pass the changed value instead. If the tz change happens after we call `monovm_initialize_preparsed`, then the notification broadcast handler would catch it and advise MonoVM about that (e.g. by calling a hypothetical `mono_update_tz_offset` function). But since that idea wasn't implemented, the next best thing is for MonoVM add its own tz change notification handler a'la the one in Xamarin.Android. It would be a bit wasteful to have two of them, but it would seem that currently it would be the only solution.

Reply:
My comment isn't about the time zone changing. It is referring to the `DateTime` that is being passed into `GetLocalUtcOffset`. The method being changed here is called from many places, not just `DateTimeOffset.Now`.

One example is `DateTimeOffset.ToLocalTime()`. If I was to run the following code:

```C#
var localTime = new DateTimeOffset(
    new DateTime(2022, 12, 29, 12, 0, 0),
    TimeSpan.Zero)
    .ToLocalTime();

Console.WriteLine(localTime.LocalDateTime);
```

That would produce different results based on _when_ the app was executed. If I ran the app in the summer, it would produce (for me in US Central):

```
12/29/2022 6:00:00 AM
```

If I ran it in the winter, it would produce:

```
12/29/2022 7:00:00 AM
```

Because we would be using the offset for `DateTime.Now` in that first second, which is not correct.

Reply:
> Is there no way in Java to get the current TZ offset without calculating the current time?

When I looked through the docs, it didn't seem like there was a direct "Current System Time Offset To UTC" API to leverage. There were multiple candidates for obtaining the value, such as `OffsetDateTime.now().getOffset().getTotalSeconds();` and `ZonedDateTime.now().getOffset().getTotalSeconds();`, but from timing via `Instant.now()` and `System.currentTimeMillis()` they seem to be similar (measuring 0 time difference). 


> One issue is that all Android apps will be paying this cost of OffsetDateTime.now() here, even if the app doesn't use DateTime.Now in C# - especially in startup code.

I think alignment with what Xamarin-Android does (which will be paying the cost of grabbing the local UTC offset via Java APIs) is preferred over the cost, plus the cost seems small.